single-host-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/singlehostclient/HttpRequestWithHostHeader.java

```diff
@@ -0,0 +1,61 @@
+package com.github.nhenneaux.resilienthttpclient.singlehostclient;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpHeaders;
+import java.net.http.HttpRequest;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+class HttpRequestWithHostHeader extends HttpRequest {
+    private final HttpRequest httpRequest;
+    private final HttpHeaders headers;
+
+    HttpRequestWithHostHeader(HttpRequest httpRequest, String hostname) {
+        this.httpRequest = httpRequest;
+        final Map<String, List<String>> map = new HashMap<>(httpRequest.headers().map());
+        map.put("host", List.of(hostname));
```

Comment:
We could potentially use `javax.ws.rs.core.HttpHeaders.HOST` constant instead.
(Beware, it has a capital `H` in the beginning. Might be an issue by the way is someone uses this header name in the map, so we have multiple "host" headers)

Reply:
there is no dependency on Jax-RS here sadly I would prefer to use such constant but as far as I know there is none in JDK?

Reply:
Indeed, it looks like there is no such a constant in the JDK.
I am still however have a small anxiety that we could have here a potential issue with different possible namings of the "host" header. In `sun.net.www.protocol.http.HttpURLConnection` the "`Host`" name is used. This however would not be enough: it is always a case-insensitve comparison is used for this maps everywhere.

Reply:
header names are case insensitive by the RFC so it should be safe.